### PR TITLE
Fully implement splittext and add test

### PR DESCRIPTION
**Pull request by amylizzle**
Slight BYOND parity break due to what looks like a bug https://www.byond.com/forum/post/2904976

### CODE REVIEW

File: Content.Tests/DMProject/Tests/Text/Splittext.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/proc/list_equal_asert(list/A, list/B)
+	ASSERT(length(A) == length(B))
+	for (var/i = 1; i <= length(A); i++)
+		ASSERT(A[i] == B[i])
```
[the remark this answers is not in the document]
Comment (amylizzle) [reply Content.Tests/DMProject/Tests/Text/Splittext.dm:4]: Completely forgot that existed

File: Content.Tests/DMProject/Tests/Text/Splittext.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/proc/list_equal_asert(list/A, list/B)
+	ASSERT(length(A) == length(B))
+	ASSERT(A ~= B)
```
Comment (wixoaGit) [inline Content.Tests/DMProject/Tests/Text/Splittext.dm:3]: This entire helper proc can be removed and replaced with `ASSERT(testX ~= testX_expected)`

File: OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs [hunk 3]
```diff
@@ -2361,22 +2361,65 @@ public static DreamValue NativeProc_oviewers(NativeProc.Bundle bundle, DreamObje
...
+            if(bundle.GetArgument(2, "Start").TryGetValueAsInteger(out start))
+                start -= 1; //1-indexed
+            if(bundle.GetArgument(3, "End").TryGetValueAsInteger(out end))
+                if(end == 0)
+                    end = text.Length;
+                else
+                    end -= 1; //1-indexed
+            bool includeDelimiters = false;
+            if(bundle.GetArgument(4, "include_delimiters").TryGetValueAsInteger(out var include_delimiters_int))
+                includeDelimiters = include_delimiters_int != 0; //idk why BYOND doesn't just use truthiness, but it doesn't, so...
```
Comment (wixoaGit) [inline OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs:2383]: Rename to `includeDelimitersInt`
```suggestion
            if(bundle.GetArgument(4, "include_delimiters").TryGetValueAsInteger(out var includeDelimitersInt))
                includeDelimiters = includeDelimitersInt != 0; //idk why BYOND doesn't just use truthiness, but it doesn't, so...
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: Content.Tests/DMProject/Tests/Text/Splittext.dm [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/proc/RunTest()
+	var/test_text = "The average of 1, 2, 3, 4, 5 is: 3"
+	var/list/test1 = splittext(test_text, " ")
+	var/list/test1_expected = list("The","average","of","1,","2,","3,","4,","5","is:","3")
+	test1 ~= test1_expected
```
Comment (wixoaGit) [inline Content.Tests/DMProject/Tests/Text/Splittext.dm:5]: These need to `ASSERT()`

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: Content.Tests/DMProject/Tests/Text/Splittext.dm (hunk 4 above)
Comment (amylizzle) [reply Content.Tests/DMProject/Tests/Text/Splittext.dm:5]: I'm an idiot

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**